include/swift/SIL/SILValue.h

```diff
@@ -626,9 +626,15 @@ llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
 /// lifetime constraints.
 struct OperandOwnership {
   enum innerty : uint8_t {
-    /// Uses of ownership None. These uses are incompatible with values that
-    /// have ownership but are otherwise not verified.
-    None,
+    /// Operands that do not use the value. They only represent a dependence
+    /// on a dominating definition and do not require liveness.
+    /// (type-dependent operands)
+    NonUse,
+
+    /// Uses that can only handle trivial values. The operand value must have
+    /// None ownership. These uses require liveness but are otherwise
```

Comment:
This is a comment for the actual commit message. Please use a different term than undef for TrivialUse operand. It isn't undefined in a compiler sense of formatting your hard drive.

Reply:
Can you add a section to this commit that distinguishes in between trivial values and trivially typed things? I think that using the enum example we talked about with a caller would work well.

Reply:
I wrote a better commit message. Just to be clear, the ownership semantics of TrivialUse literally have undefined ownership semantics, which is the whole reason we want the compiler to enforce that they can't operate on values with ownership semantics (aka nontrivial values). Contrast this with things like InstantaneousUse or BitwiseEscape, which just don't care about ownership (i.e. they have no ownership semantics). We only get undefined behavior when the compiler does not enforce code that has undefined semantics.
https://github.com/apple/swift/pull/35248/commits/ce2a7cfe0edeb179589c99a2e2e9b0b90b06a819

Reply:
The enum cases shouldn't have much extra commentary so the enum itself is readable. But I clarified all the comments, and since "trivial value" isn't defined anywhere, I added an introduction in the OwnershipKind doc comment.
https://github.com/apple/swift/pull/35248/commits/ce2a7cfe0edeb179589c99a2e2e9b0b90b06a819

I find it grammatically impossible not to use the term "trivial values" when writing about ownership because the term "None values" doesn't parse. If I ever argued against using the word trivial, I deeply apologize for that.